swap/tests/testutils/mod.rs

```diff
@@ -302,7 +304,7 @@ impl TestContext {
     }
 }
 
-pub async fn setup_test<T, F>(testfn: T)
+pub async fn setup_test<T, F>(override_config: Option<Config>, testfn: T)
```

Comment:
If you change `override_config` to be of type `Option<O>` and add `where O: GetConfig` then you don't have to call `get_config()` on the outside but can just call it in here by doing:

```
let config = override_config.unwrap_or(config::Regtest)::get_config()
```

You might need to change the associated function to a method for this to work (i.e. make `get_config` take `self`)

Reply:
no need to make it a method, just need to call `O::get_config()` :)